src/amo/imageUtils.js

```diff
@@ -1,7 +1,25 @@
 import fallbackIcon from 'amo/img/icons/default.svg';
 
-export function getAddonIconUrl(addon) {
-  return addon ? addon.icon_url : fallbackIcon;
+//
+// returns icon url from icons objects,
+// or returns icon_url value if icons does not exists
+//
+// fallback icon will be returned if the addon has no icon.
+//
+// iconSize values are 32 | 64 | 128
+export function getAddonIconUrl(addon, size) {
+  const iconSizes = [32, 64, 128];
+  const iconSizeIsNotValid = size && !iconSizes.includes(size);
+
+  if (iconSizeIsNotValid) {
+    throw new Error(`size must be one of: ${iconSizes.toString()}`);
+  }
+
+  const defaultIconSize = iconSizes[1];
```

Comment:
We don't want to throw an error if the requested size isn't valid really, it doesn't help - if it's not valid it won't be in `addon.icons` and it'll nicely fallback.  (so you don't need to define a list of `iconSizes` here either)

Reply:
You are right, counterproductive extra logic.
I removed it.